lib/jsdom/living/nodes/HTMLSelectElement-impl.js

```diff
@@ -82,19 +83,69 @@ class HTMLSelectElementImpl extends HTMLElementImpl {
       selected.forEach((option, index) => {
         option._selectedness = index === selected.length - 1;
       });
+      this._selectedOption = selected[selected.length - 1];
     }
   }
 
+  _updateSelected(child) {
+    if (this.hasAttributeNS(null, "multiple")) {
+      return;
+    }
+    if (child.hasAttributeNS(null, "disabled")) {
+      return;
+    }
+    const optArr = this._toOptionArray(child);
+    if (!optArr.length) {
+      return;
+    }
+    const selectedOptions = optArr.filter(opt => opt._selectedness);
+    // If any selected is last selected
+    if (selectedOptions.length) {
+      // pushed at least one element with selected=true, reset
+      this._askedForAReset();
+      return;
+    }
+
+    // No selection so first option is selected
+    if ((!this._selectedOption || !this._selectedOption._selectedness) && this._displaySize === 1) {
+      this._selectedOption = optArr[0];
+      this._selectedOption._selectedness = true;
+    }
+  }
+
+  _toOptionArray(child) {
+    const array = [];
+    if (child._localName === "option") {
+      array.push(child);
```

Comment:
It might be nicer to check for disabledness here, and then not need to check for it on line 94. Then the two branches of the toOptionArray function are symmetric.

Reply:
Have made the change to check for disabledness in the `toOptionArray` here so the branches are symmetric and its consistent. Have left the check on line 94 as :    
1) Don't want to have to allocate a new array in here needlessly    
2) That also checks if the child is an optiongroup and the entire group is disabled which I am not currently doing here but could add.     
Not particularly passionate about any of that so if you also want it remove from the newly named `_selectednessSettingAlgorithmWithKnownNewAddition` just let me know.